device-posture/device-posture-change-event.https.html

```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script>
+'use strict';
+
+// REFERENCE https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/http/tests/inspector-protocol/device-posture/device-posture-change-event.js
+promise_test(async (t) => {
+  const watcher = new EventWatcher(t, navigator.devicePosture, ['onchange']);
```

Comment:
> Look at other EventWatcher users do ("foo" and "onfoo" are different).

Unfortunately this tip is not very useful. The variation between different EventWatcher's is huge. They wait load, message, securitypolicyviolation, activate, reading, error, ...

I think I should look https://w3c.github.io/device-posture/ . I was confused as it talks lot about "onchange" (and that test seems to works (partially) with "onchange" word), but now I found this chapter https://w3c.github.io/device-posture/#the-onchange-attribute-handle-posture-changes which says the event type is "change"

After onchange => change I get into problems with this line
`await test_driver.set_device_posture(posture);`
If I have "await" in there I get "Not expecting event, but got change event expected true got false" error. 
Also I have to change the "await watcher;" line
With onchange => change and with below kind of look code seems to work

```
promise_test(async (t) => {
  const watcher = new EventWatcher(t, navigator.devicePosture, ['change']);
  const postures = ['folded', 'continuous', 'folded'];
  for (const posture of postures) {
    test_driver.set_device_posture(posture);
    await watcher.wait_for('change');
    assert_equals(posture, navigator.devicePosture.type);
  }
}, 'Device posture test: device-posture-change-event.js.');
```

Reply:
> Unfortunately this tip is not very useful. The variation between different EventWatcher's is huge. They wait load, message, securitypolicyviolation, activate, reading, error, ...
> 
> I think I should look https://w3c.github.io/device-posture/ . I was confused as it talks lot about "onchange" (and that test seems to works (partially) with "onchange" word), but now I found this chapter https://w3c.github.io/device-posture/#the-onchange-attribute-handle-posture-changes which says the event type is "change"

I did not write a more detailed explanation because this shouldn't have been a surprise to you by now. You've written code that does `object.onfoo = ...` as well as `object.addEventListener('foo')`, and you've dealt with EventWatcher in Generic Sensors and Device Orientation in the past months. The Generic Sensors spec has [a section listing event handlers](https://w3c.github.io/sensors/#event-handlers), and the entire Device Orientation API is based around events, so a `git grep EventWatcher` in the corresponding directories in WPT should have given you more information to support this work.


> After onchange => change I get into problems with this line `await test_driver.set_device_posture(posture);` If I have "await" in there I get "Not expecting event, but got change event expected true got false" error. Also I have to change the "await watcher;" line With onchange => change and with below kind of look code seems to work
> 
> ```
> promise_test(async (t) => {
>   const watcher = new EventWatcher(t, navigator.devicePosture, ['change']);
>   const postures = ['folded', 'continuous', 'folded'];
>   for (const posture of postures) {
>     test_driver.set_device_posture(posture);
>     await watcher.wait_for('change');
>     assert_equals(posture, navigator.devicePosture.type);
>   }
> }, 'Device posture test: device-posture-change-event.js.');
> ```

This is a variation on a familiar theme (e.g. `test_driver.update_virtual_sensor()` and `watcher.wait_for('reading')`), so similar advice applies.

```js
// Option 1.
await Promise.all([
  watcher.wait_for('change'),
  test_driver.set_device_posture(posture)
]);

// Option 2, just a more verbose version of option 1.
const watchPromise = watcher.wait_for('change');
await test_driver.set_device_posture(posture);
await watchPromise;
```

Reply:
Fixed in https://github.com/JuhaVainio/WPT/pull/4/commits/291385c15ed5cf39ad5a37182bcc1994ee799744